perceval/backends/github.py

```diff
@@ -68,6 +69,22 @@ def __init__(self, owner=None, repository=None, token=None,
         self.owner = owner
         self.repository = repository
         self.client = GitHubClient(owner, repository, token, base_url)
+        self._users = {}  # internal users cache
+
+    def _add_users(self, issue):
```

Comment:
Yes, I don't like also to modify dicts inside external functions, but all the logic in this method is the assign logic for user and assignee. If we move it to the main function, we can just remove this function and _add_users_from_cache also, and have a bigger fetch functions. If you are more comfortable with this approach, I will do that.